crates/primitives/src/net.rs

```diff
@@ -47,6 +47,29 @@ impl NodeRecord {
         Self::new(addr, id)
     }
 
+    /// Converts the `address` into an [`Ipv4Addr`] if the `address` is a mapped
+    /// [Ipv6Addr](std::net::Ipv6Addr).
+    ///
+    /// Returns `true` if the address was converted.
+    ///
+    /// See also [std::net::Ipv6Addr::to_ipv4_mapped]
+    pub fn convert_ipv4_mapped(&mut self) -> bool {
```

Comment:
Instead of doing it this way - should we do it by default on the constructor?

Reply:
since it's common to instantiate the struct directly (all fields pub) we should just keep this separately imo